larrecodnn/ImagePatternAlgs/Tensorflow/Modules/waveformroifinder.fcl

```diff
@@ -3,8 +3,50 @@ BEGIN_PROLOG
 standard_waveformroifinder:
 {
     module_type: "WaveformRoiFinder"
+    NPlanes:         3
+    WaveformSize:    6000
+    RawProducerLabel:   ""
     WireProducerLabel:  "caldata:dataprep"
-    WaveformRecog: {
+
+    WaveformRecogPlane0: {
```

Comment:
There are a few issues here:

- The number of planes should be determined by the geometry service (as mentioned above)
- The `WaveformSize` is stated both here and in the configurations of the tools; is it necessary in both places?
- Why provide a `RawProducerLabel` default of the empty string?  That is the default value in the C++ file anyway and it is not necessary to provide it here.
- If you think it is necessary to have a tool for each plane, then it would be much safer (and simpler in the C++ file) to use a sequence of configurations instead of configurations with hard-coded indices:
    ```.jsonnet
    WaveformRecogs: [
        { /* config for first tool */ },
        { /* config for second tool */ },
        { /* config for third tool */ }
    ]
    ```
- The configurations for each of the tools appear to be identical.  So do you really need multiple tools for this module?  If you decide you do, you should introduce a FHiCL parameter that can be used elsewhere:
    ```.jsonnet
    tool_config: {
      NNetModelFile: "CnnModels/lightmodel112.pb"
      ...
    }
    # And then in the configuration body:
    WaveformRecogs: [
      @local::tool_config,
      @local::tool_config,
      @local::tool_config
    ]
    ```